samza-kafka/src/main/scala/org/apache/kafka/clients/consumer/KafkaConsumerConfig.java

```diff
@@ -0,0 +1,180 @@
+      LOG.info("setting default key serialization for the consumer(for {}) to ByteArrayDeserializer", systemName);
+      consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+    }
+    if (!config.containsKey(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG)) {
+      LOG.info("setting default value serialization for the consumer(for {}) to ByteArrayDeserializer", systemName);
+      consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+    }
+
+    // NOT SURE THIS IS NEEDED TODO
+    String maxPollRecords =
+        subConf.get(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, KAFKA_CONSUMER_MAX_POLL_RECORDS_DEFAULT);
+    consumerProps.setProperty(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, maxPollRecords);
+
+    // put overrides
+    consumerProps.putAll(injectProps);
+
+    return new KafkaConsumerConfig(consumerProps);
+  }
+
+  // group id should be unique per job
+  static String getConsumerGroupId(Config config) {
+    JobConfig jobConfig = new JobConfig(config);
+    Option<String> jobIdOption = jobConfig.getJobId();
+    Option<String> jobNameOption = jobConfig.getName();
+    return (jobNameOption.isDefined() ? jobNameOption.get() : "undefined_job_name") + "-" + (jobIdOption.isDefined()
+        ? jobIdOption.get() : "undefined_job_id");
+  }
+
+  // client id should be unique per job
+  public static String getConsumerClientId(Config config) {
+    return getConsumerClientId(CONSUMER_CLIENT_ID_PREFIX, config);
+  }
+  public static String getProducerClientId(Config config) {
+    return getConsumerClientId(PRODUCER_CLIENT_ID_PREFIX, config);
+  }
+  public static String getAdminClientId(Config config) {
+    return getConsumerClientId(ADMIN_CLIENT_ID_PREFIX, config);
+  }
+
+  private static String getConsumerClientId(String id, Config config) {
+    if (config.get(JobConfig.JOB_NAME()) == null) {
+      throw new ConfigException("Missing job name");
+    }
+    String jobName = config.get(JobConfig.JOB_NAME());
+    String jobId = (config.get(JobConfig.JOB_ID()) != null) ? config.get(JobConfig.JOB_ID()) : "1";
+
+    return String.format("%s-%s-%s", id.replaceAll("[^A-Za-z0-9]", "_"), jobName.replaceAll("[^A-Za-z0-9]", "_"),
+        jobId.replaceAll("[^A-Za-z0-9]", "_"));
+  }
+
+  /**
+   * Settings for auto.reset in samza are different from settings in Kafka (auto.offset.reset) - need to convert
+   * "largest" -> "latest"
+   * "smallest" -> "earliest"
+   * "none" -> "none"
+   * "none" - will fail the kafka consumer, if offset is out of range
+   * @param properties All consumer related {@link Properties} parsed from samza config
+   * @return String representing the config value for "auto.offset.reset" property
+   */
+  static String getAutoOffsetResetValue(Properties properties) {
+    String autoOffsetReset = properties.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, KAFKA_OFFSET_LATEST);
```

Comment:
1. Can you please share the rationale why we want to allow users to define configuration `auto.reset.offset` with predefined kafka constants  directly? 
2. Previously, `samza`  did not allow users to specify value anything other than `largest`, `smallest` for the configuration `auto.reset.offset`. Now if we want to allow users to specify the predefined kafka  `auto.reset.offset` configurations, can we update the configuration table documentation in accordance with this implementation.

Reply:
I agree about the utility of allowing users to set auto.reset.offset, but doesn't make sense to allow using Samza's config values for Kafka's config keys. We should to remove this method and pass the user provided consumer config as-is.